### PR TITLE
feat: Add Twitch clip and marker actions, refactor stream title

### PR DESCRIPTION
**Pull request by Destroy666x**
Adds:
- https://dev.twitch.tv/docs/api/reference/#create-stream-marker
- https://dev.twitch.tv/docs/api/reference/#create-stream-marker

and refactors `text` to `streamTitle`. Also small change of `.gitignore` for VSCode as that's my editor of choice and what I currently have in `.vscode` as a C++ beginner there is definitely not worth commiting.

WIP until I build and test tommorow.

### CODE REVIEW

**Comment by Destroy666x** [comment]
Closes https://github.com/WarmUpTill/SceneSwitcher/issues/873

File: src/macro-external/twitch/macro-action-twitch.cpp [hunk 1]
```diff
@@ -131,8 +175,10 @@ bool MacroActionTwitch::Save(obs_data_t *obj) const
 	obs_data_set_int(obj, "action", static_cast<int>(_action));
 	obs_data_set_string(obj, "token",
 			    GetWeakTwitchTokenName(_token).c_str());
-	_text.Save(obj, "text");
+	_streamTitle.Save(obj, "streamTitle");
```
Comment (WarmUpTill) [inline src/macro-external/twitch/macro-action-twitch.cpp:178]: Just to mention it:
This will break your existing macros.

File: .gitignore
Comment (WarmUpTill) [inline .gitignore:1]: If you don't mind please separate this out to its own commit.

**Comment by WarmUpTill** [comment]
Looks good to me, besides the minor formatting issues :)
Let me know if you need help in resolving these.

File: src/macro-external/twitch/macro-action-twitch.cpp (hunk 1 above)
Comment (Destroy666x) [reply src/macro-external/twitch/macro-action-twitch.cpp:178]: Yeah, changed it as that was not released yet and your recommendation when I asked, so hopefully won't be a problem (no idea myself :P)

File: .gitignore
Comment (Destroy666x) [reply .gitignore:1]: Sure, will do

**Comment by Destroy666x** [comment]
> Looks good to me, besides the minor formatting issues :) Let me know if you need help in resolving these.

What are you/would you recommend using? VSCode extension had something called "clangFormat" which I see you're also using, but that made very weird looking spacing changes which were also often different from spacing in the file, so I disabled it. But maybe they just seemed weird as I've never used modern C++ and that's somehow the (confusing IMO) standard, since I see in the build that clangFormat is complaining  ¯\_(ツ)_/¯ Or maybe I need to further configure the plugin somehow. Is there some config file to follow like there would be e.g. `.php-cs-fixer.php` in PHP?

**Comment by Destroy666x** [comment]
And one small issue I noticed while testing from builds here - the placeholders in text fields are not shown. Do they have some sort of text limit?

And another question BTW - is there some sort of field validation implemented in the plugin? E.g. displaying some kind of error  on save when text input is too long. Couldn't see any obvious one when I checked. But maybe it's something more manual, like if conditions in `save` or similar function.

**Comment by WarmUpTill** [comment]
>What are you/would you recommend using?

VS code should work fine.
Although I personally prefer using Visual Studio for development on Windows.

Both should be able to work with .clang-format.

>And one small issue I noticed while testing from builds here - the placeholders in text fields are not shown. Do they have some sort of text limit?

To be honest I am not sure what exactly you are referring to here.

>And another question BTW - is there some sort of field validation implemented in the plugin?

Using the maxLength property is one option.
Displaying an error would have to be done manually.
You can use this helper if you like:
https://github.com/WarmUpTill/SceneSwitcher/blob/fc1460e59248ce7350ad1611595fc5aedd3e1fda/src/utils/utility.hpp#L107

**Comment by Destroy666x** [comment]
I mean the default value that you had as `AdvSceneSwitcher.enterText` (Something like `-- Enter Text--`) before as default value in the `.hpp` field definition. For some reason the longer names weren't displayed in previous build. I'll check the shorter in these

**Comment by Destroy666x** [comment]
Tooltips work now (so I guess it was length) and the rest looks ok too, I guess it's mergeable now?
